test/results_test.exs

```diff
@@ -58,13 +68,118 @@ defmodule Wanda.ResultsTest do
                %ExecutionResult{
                  execution_id: ^execution_id,
                  group_id: ^group_id,
-                 payload: %{
-                   "result" => "passing",
-                   "check_results" => [_ | _]
-                 }
+                 status: :running,
+                 payload: %{},
+                 targets: [
+                   %{
+                     agent_id: ^agent_id_1,
+                     checks: ^checks_1
+                   },
+                   %{
+                     agent_id: ^agent_id_2,
+                     checks: ^checks_2
+                   }
+                 ]
                }
              ] = Repo.all(ExecutionResult)
     end
+
+    test "should raise an error when trying to create an already existing execution" do
+      [
+        %ExecutionResult{execution_id: execution_id, group_id: group_id},
+        %ExecutionResult{}
+      ] = insert_list(2, :execution_result)
+
+      assert_raise Ecto.ConstraintError, fn ->
+        Results.create_execution_result!(execution_id, group_id, build_list(2, :target))
+      end
+
+      assert 2 =
+               ExecutionResult
+               |> Repo.all()
+               |> length()
+    end
+  end
+
+  describe "Completing an Execution" do
+    test "should correctly complete a running execution" do
+      %ExecutionResult{
+        execution_id: execution_id,
+        group_id: group_id,
+        status: :running,
```

Comment:
We can remove this `status` and `payload` here I guess. They are not used

Reply:
What I want to express here by pattern matching `status: :running` and `payload: %{}` is that we are starting with a running execution (status running and empty payload) and ending up with that same execution being completed and with some expected payload (later in the assertion).

Reply:
But you already know this when you add `make_running`. You are simply testing your own helper methods and factories.

Reply:
Alright changed here https://github.com/trento-project/wanda/pull/60/commits/bde3ce30348124a8ad986edf41320737046bc695